mentat/session_stream.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+from contextvars import ContextVar
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from typing import Any, AsyncGenerator, Dict, List, cast
+from uuid import UUID, uuid4
+
+from .broadcast import Broadcast
+
+logger = logging.getLogger()
+
+_SESSION_STREAM: ContextVar[SessionStream] = ContextVar("mentat:session_stream")
+
+
+def get_session_stream():
+    return _SESSION_STREAM.get()
+
+
+def set_session_stream(session_stream: SessionStream):
+    return _SESSION_STREAM.set(session_stream)
```

Comment:
Probably can remove these get/set methods and just make `_SESSION_STREAM` a public variable